docs/gcp.md

```diff
@@ -0,0 +1,166 @@
+steps:
+  - name: 'gcr.io/cloud-builders/docker'
+    entrypoint: 'bash'
+    args:
+      - '-c'
+      - |
+        docker pull us-central1-docker.pkg.dev/$PROJECT_ID/swift/server:latest || exit 0
+  - name: 'gcr.io/cloud-builders/docker'
+    args:
+      - build
+      - -t
+      - us-central1-docker.pkg.dev/$PROJECT_ID/swift/server:$SHORT_SHA
+      - -t
+      - us-central1-docker.pkg.dev/$PROJECT_ID/swift/server:latest
+      - .
+      - --cache-from
+      - us-central1-docker.pkg.dev/$PROJECT_ID/swift/server:latest
+  - name: 'gcr.io/cloud-builders/docker'
+    args:
+      ['push', 'us-central1-docker.pkg.dev/$PROJECT_ID/swift/server:$SHORT_SHA']
+  - name: 'gcr.io/cloud-builders/gcloud'
+    args:
+      - run
+      - deploy
+      - swift-service
+      - --image=us-central1-docker.pkg.dev/$PROJECT_ID/swift/server:$SHORT_SHA
+      - --port=8080
+      - --region=us-central1
+      - --memory=512Mi
+      - --platform=managed
+      - --allow-unauthenticated
+      - --min-instances=0
+      - --max-instances=5
+images:
+  - 'us-central1-docker.pkg.dev/$PROJECT_ID/swift/server:$SHORT_SHA'
+  - 'us-central1-docker.pkg.dev/$PROJECT_ID/swift/server:latest'
+timeout: 1800s
+```
+
+### The steps in detail
+
+1. pull the latest image from the Artifact Registry to retrieve cached layers
+2. built the image with `$SHORT_SHA` and `latest` tag
+3. push the image to the Artifact Registry
+4. deploy the image to Cloud Run
+
+`images` specifies the build images to store in the restristry. The default
+`timeout` is 10 minutes, so we'll need to increase it for Swift builds. We use
+`8080` as the default port here, it's recommended to remove this line and have
+the server listen on `$PORT`.
+
+## Deployment
+
+![cloud build trigger settings and how to connect a code repository](../images/gcp-connect-repo.png)
+
+Push all files to a GitHub or BitBucket (the only two supported providers right
+now) and head to
+[Cloud Build Triggers](https://console.cloud.google.com/cloud-build/triggers)
+and click "Create Trigger":
+
+1. Add a name and description
+2. Event: "Push to a branch" is active
+3. Source: "Connect New Repository" and authorize with your code provider, add
+   the repository where your swift server code is hosted
+4. Configuration: "Cloud Build configuration file" / Location: Repository
+5. Advanced:
+   [Substitution variables](https://cloud.google.com/cloud-build/docs/configuring-builds/substitute-variable-values):
+   if you use environment variables for example to connect to a database or 3rd
+   party services, you can set the values here. You can also control the build
+   arguments such as memory, region etc.
+6. "Create"
+
+In the Trigger overview page, you should see your new "swift-service" trigger.
+Click on "RUN" on the right to start the trigger manually from the `main`
+branch. With a simple Hummingbird project the build takes about 7-8 minutes.
+Vapor takes about 25 minutes. After a successful build you should see the
```

Comment:
cc @tomerd @Lukasa this is due to having to build BoringSSL twice, once for NIOSSL and once for Swift Crypto. Reducing that with a single `SwiftCryptoExtras` that works for both would be great

Reply:
A basic Hummingbird project doesn't include either NIOSSL or SwiftCrypto, so 7 or 8 mins feels quite long.

Reply:
I'm guessing the default machines are quite slow. On GH actions a large Vapor project with no caching including Soto and JWT (so 3 or 4 copies of BoringSSL) takes about 12 minutes to build the container

Reply:
Added a recommendation from the Vapor channel